### PR TITLE
Fix bug 1396080: Encode non-encoded querystring

### PR DESCRIPTION
**Pull request by mathjazz**
@jotes r?

### CODE REVIEW

File: pontoon/base/static/js/translate.js [hunk 1]
```diff
@@ -4015,7 +4015,16 @@ var Pontoon = (function (my) {
...
       }
 
-      return decodeURIComponent(results[2].replace(/\+/g, " "));
+      var encodedURI = results[2].replace(/\+/g, " ");
+
+      try {
+        return decodeURIComponent(encodedURI);
+
+      // If querystring not encoded, we need to encode it first
+      } catch (e if e instanceof URIError) {
```
Comment (jotes) [inline pontoon/base/static/js/translate.js]: btw. Did you consider to add raygun.io to the frontend?

Comment (julen) [inline pontoon/base/static/js/translate.js]: This inline conditional catch syntax [is not standard and its use is discouraged](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#Conditional_catch_clauses). You should check for `URIError` in a separate conditional block.

**Comment by jotes** [comment]
thx @julen. I've seen that some of sites use the same exception handling and I assumed that if it works in Fx (that's the browser in which this bug was verified) that means Chrome also will work.
TIL: that syntax is supported only in Firefox and not available in Chrome/Safari/Opera.
@mathjazz @adngdb I would add a bug for javascript linter in pontoon.

File: pontoon/base/static/js/translate.js (hunk 1 above)
Comment (mathjazz) [reply pontoon/base/static/js/translate.js]: Thanks. I had no idea. And I actually found this syntax in the exact same article. :/

**Comment by mathjazz** [comment]
Updated.

@jotes I support the idea of a JS linter.

**Pull request merged**